src/Storages/MergeTree/MergeTreeIndexIVFFlat.cpp

```diff
@@ -0,0 +1,260 @@
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <string>
+#include <Storages/MergeTree/MergeTreeIndexIVFFlat.h>
+
+#include <IO/ReadBuffer.h>
+#include <IO/WriteBuffer.h>
+
+#include <Parsers/ASTFunction.h>
+
+#include <Poco/Logger.h>
+#include <base/logger_useful.h>
+
+#include "Core/Field.h"
+#include "MergeTreeIndices.h"
+#include "base/types.h"
+#include <Common/FieldVisitorsAccurateComparison.h>
+
+#include <faiss/IndexFlat.h>
+#include <faiss/IndexIVF.h>
+#include <faiss/IndexIVFFlat.h>
+#include <faiss/MetricType.h>
+#include <faiss/impl/io.h>
+#include <faiss/index_io.h>
+
+namespace DB
+{
+
+namespace detail {
+    class WriteBufferFaissWrapper : public faiss::IOWriter {
```

Comment:
Add some comments about this class (maybe on next stage). Just a short description of why it should exist.